<!--
  Author: Mahimana Bhatt
  Email: mbhatt@wpi.edu

  Team Capricorn
  NASA SPACE ROBOTICS CHALLENGE -->

<launch>
  <arg name="robot_name" default="small_scout_1" />   
  <arg name="rviz" default="true" />

  <arg name="namespace" value="/$(arg robot_name)/laser" />

  <group ns="$(arg namespace)">
  <!-- Odometry -->
  <node pkg="rtabmap_ros" type="icp_odometry" name="stereo_odometry">
      <param name="frame_id" type="string" value="$(arg robot_name)_base_footprint"/>
      <param name="approx_sync" type="bool" value="true"/>
      <param name="queue_size" type="int" value="5"/>
      <param name="odom_frame_id" type="string" value="odom"/>
      <param name="publish_tf"  type="bool"   value="true"/>
      <param name="Icp/MaxCorrespondenceDistance" type="string" value="0.1"/>
      <param name="Reg/Force3DoF" type="string" value="true"/>
  </node>     
  </group>  

  <!-- Visualisation RVIZ --> 
  <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rviz" args="-d $(find utils)/rviz/compare_odom.rviz"/>
  <!-- <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rviz" args="-d $(find rtabmap_ros)/launch/config/demo_stereo_outdoor.rviz"/> -->

</launch>